### PR TITLE
[frontend] Move constants to the classes they belong to

### PR DESCRIPTION
**Pull request by bgeuken**
We just shouldn't define constants via initializers. What else could I
say?;-)

### CODE REVIEW

File: src/api/app/models/bs_request.rb [hunk 1]
```diff
@@ -32,6 +32,8 @@ class SaveError < APIException
 
   FINAL_REQUEST_STATES = %w(accepted declined superseded revoked)
 
+  VALID_REQUEST_STATES = [:new, :deleted, :declined, :accepted, :review, :revoked, :superseded]
```
Comment (coolo) [inline src/api/app/models/bs_request.rb]: as I'm curious: why define one states as %w and one as symbol array?

Comment (bgeuken) [reply src/api/app/models/bs_request.rb]: Probably because there was no coding style. And I think we still don't have one arrays.

I just copied this over;-)

Comment (coolo) [reply src/api/app/models/bs_request.rb]: you mean @Ana06 has no cop for that? sad!

Comment (bgeuken) [reply src/api/app/models/bs_request.rb]: We decide such things together;-)

There is a cop available. Either we didn't enable it yet, or it allows both ways by default.

Comment (Ana06) [reply src/api/app/models/bs_request.rb]: @coolo your wish is my command: https://github.com/openSUSE/open-build-service/pull/4004

Comment (Ana06) [reply src/api/app/models/bs_request.rb]: @bgeuken it was not enabled. But nice to have it. The `%w()` is the default one. Maybe you can change it here :pray:

File: src/api/app/models/review.rb [hunk 2]
```diff
@@ -5,6 +5,8 @@ class NotFoundError < APIException
     setup 'review_not_found', 404, 'Review not found'
   end
 
+  VALID_REVIEW_STATES = [:new, :declined, :accepted, :superseded, :obsoleted].freeze
```
Comment (Ana06) [inline src/api/app/models/review.rb]: why did you add freeze here and not in the other one? :thinking:

File: src/api/app/models/bs_request.rb (hunk 1 above)
Comment (bgeuken) [reply src/api/app/models/bs_request.rb]: @Ana06 In principle yes, but I actually prefer the array syntax. So let's keep this discussion split from this changes.
It's two different things anyway:-)

File: src/api/app/models/review.rb (hunk 2 above)
Comment (bgeuken) [reply src/api/app/models/review.rb]: Seems I forgot that. I'll fix it now.

**Comment by Ana06** [comment]
@coolo 

> as I'm curious: why define one states as %w and one as symbol array?

you got us confused! `%w` is only for strings!!!!! :rofl:

**Comment by coolo** [comment]
my pleasure :)
